src/main/php/PHPMD/Renderer/CheckStyleRenderer.php

```diff
@@ -51,7 +51,7 @@ public function renderReport(Report $report): void
         foreach ($report->getRuleViolations() as $violation) {
             $fileName = $violation->getFileName();
 
-            if ($this->fileName !== $fileName) {
+            if ($this->fileName !== $fileName && $fileName) {
```

Comment:
Actually if it happens, this will then merge the error with the previous file and if it's the first one, we'll have a `</file`> with no paired `<file>` so instead of that, I would provide a fallback value.
```suggestion
            if ($this->fileName !== $fileName) {
```